docs/tutorial/7-snapping.md

```diff
@@ -0,0 +1,610 @@
+
+Calls made on thread1: 100000
+Calls made on thread2: 1
+```
+
+- Debug output:
+```
+= Seccomp =
+Time: Mar 27 20:29:40
+Log: auid=1000 uid=1000 gid=1000 ses=119 subj=snap.rt-app.edf pid=37555 comm="edf" exe="/snap/rt-app/x1/bin/edf" sig=0 arch=c000003e 314(sched_setattr) compat=0 ip=0x78846c90325d code=0x7ffc0000
+Syscall: sched_setattr
+Suggestion:
+* add 'process-control' to 'plugs'
+```
+
+The [seccomp] message indicates that the application makes a [sched_setattr] system call.
+As confirmed in the source code, this occurs inside the `thread_start` function.
+The debug tool suggests adding the [process-control interface][proc_ctr_iface].
+To understand interfaces, plugs, and slots, refer to the [interface management document][iface_mgmt].
+
+To grant the necessary permission, add `process-control` to the `plugs` field under the `edf` app definition:
+
+```yaml
+  edf:
+    command: bin/edf
+    plugs:
+      - process-control
+```
+
+Rebuild and reinstall the Snap, then connect the interface:
+
+```{note}
+You may see some logs on the `snappy-debug` terminal regarding `lxd` during the `snapcraft` build, those can be safely ignored.
+```
+
+```shell
+sudo snap connect rt-app:process-control
+```
+
+Listing Snap connections confirms the manual connection:
+
+```{terminal}
+   :input: snap connections rt-app
+   :user: ubuntu
+   :host: ubuntu
+   :dir: ~/tutorial
+
+Interface        Plug                    Slot              Notes
+process-control  rt-app:process-control  :process-control  manual
+```
+
+#### Analyzing `fifo` and `lock` applications
+
+Repeat the same process done for `cfs` and `edf` applications on this one. 
+Like occurred for `cfs`, when running `fifo` application no reports were generated by `snappy-conf`.
+The same happens for the `lock` application, no reports were generated.
+
+
+#### Analyzing `thread-affinity` application
+
+Repeating the process for `thread-affinity` something different happens on the logs:
+
+- Application output:
+```{terminal}
+   :input: rt-app.thread-affinity
+   :user: ubuntu
+   :host: ubuntu
+   :dir: ~/tutorial
+
+thread1 priority: 0
+thread2 priority: 0
+Calls made on thread1: 100000
+Calls made on thread2: 1
+```
+
+- Debug output:
+```
+= Seccomp =
+Time: Mar 27 21:04:22
+Log: auid=1000 uid=1000 gid=1000 ses=119 subj=snap.rt-app.thread-affinity pid=40684 comm="thread-affinity" exe="/snap/rt-app/x2/bin/thread-affinity" sig=0 arch=c000003e 203(sched_setaffinity) compat=0 ip=0x733af5a4d6e1 code=0x7ffc0000
+Syscall: sched_setaffinity
+Suggestion:
+* ignore the denial if the program otherwise works correctly (unconditional sched_setaffinity is often just noise)
+```
+
+We see a seccomp report, but no suggested interface to use.
+Instead there is a message suggesting to just ignore the denial report.
+This happens because there are some denials coming from the security policies defined in the snap sandboxing, which potentially does not affect the functionality of an application.
```

Comment:
denials (plural) do not ...
```suggestion
This happens because there are some denials coming from the security policies defined in the snap sandboxing, which potentially do not affect the functionality of an application.
```